src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -568,20 +570,20 @@ private bool ReportNullableAssignmentIfNecessary(BoundExpression value, TypeSymb
 
             // Report warning converting null literal to non-nullable reference type.
             // target (e.g.: `object x = null;` or calling `void F(object y)` with `F(null)`).
-            bool reportNullLiteralAssignmentIfNecessary()
+            bool reportNullLiteralAssignmentIfNecessary(BoundExpression expr)
```

Comment:
Other than making this a candidate for a `static` local function when we get them, did this actually add any value? Not criticizing, just making sure I'm not missing something.

Reply:
Passing the expression as an argument allows null analysis to verify that `value` is not null in this method (see [https://github.com/dotnet/roslyn/pull/31719](https://github.com/dotnet/roslyn/pull/31719)).

---
In reply to: [241615378](https://github.com/dotnet/roslyn/pull/31771#discussion_r241615378) [](ancestors = 241615378)